retrieval_service/datastore/providers/cloudsql_postgres.py

```diff
@@ -330,17 +330,34 @@ async def get_amenity(self, id: int) -> Optional[models.Amenity]:
         return res
 
     async def amenities_search(
-        self, query_embedding: list[float], similarity_threshold: float, top_k: int
+        self,
+        query_embedding: list[float],
+        similarity_threshold: float,
+        top_k: int,
+        filter_time: Optional[str],
+        filter_day: Optional[str],
     ) -> list[models.Amenity]:
+
+        filter_time_datetime = None
+        filter_query = "WHERE "
+        if filter_time and filter_day:
+            start_hour = filter_day + "_start_hour"
+            end_hour = filter_day + "_end_hour"
```

Comment:
Should we validate the possible options that filter_day should be?

Reply:
Added the filter in route.py. Seems like theres where most endpoint input validations were made.